jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -209,19 +209,26 @@ YUI.add('user-profile-model-list', function() {
           </div>
         );
       }
-      const list = this.state.modelList;
-      if (!list || list.length === 0) {
-        return null;
-      }
-      const rows = list.map(this._generateRow);
       const props = this.props;
+      const acl = props.acl;
       let createNewButton;
-      if (props.canCreateNew) {
+      if (acl && acl.canAddModels()) {
```

Comment:
@frankban I wondered the same about whether `isReadOnly` is needed, after I looked that the `acl.js` code. `isReadOnly` looks at model-level access, whereas `canAddModels` looks at controller-level access. They don't really seem to be related as I'd initially thought.

As to the defensive check: we have two routes here:

1. Make `acl` a required prop, in which case the defensive check isn't needed.
2. Leave it as it is, an optional prop, and leave the defensive check in.

In general I feel like we have way, way too many required props, so my default preference is for optional props.